fuzz/fuzz_targets/fuzz_int_set.rs

```diff
@@ -0,0 +1,673 @@
+    }
+}
+
+/* ### Extend ### */
+
+struct ExtendOp(Vec<u32>);
+
+impl ExtendOp {
+    fn parse_args(data: &[u8]) -> (Option<Box<dyn Operation>>, &[u8]) {
+        let (Some(values), data) = read_u32_vec(data) else {
+            return (None, data);
+        };
+        (Some(Box::new(Self(values))), data)
+    }
+}
+
+impl Operation for ExtendOp {
+    fn operate(&self, input: Input, _: Input) {
+        input.int_set.extend(self.0.iter().copied());
+        input.btree_set.extend(self.0.iter().copied());
+    }
+
+    fn size(&self, length: usize) -> usize {
+        (length.ilog2() as usize) * self.0.len()
+    }
+}
+
+/* ### Extend Unsorted ### */
+
+struct ExtendUnsortedOp(Vec<u32>);
+
+impl ExtendUnsortedOp {
+    fn parse_args(data: &[u8]) -> (Option<Box<dyn Operation>>, &[u8]) {
+        let (Some(values), data) = read_u32_vec(data) else {
+            return (None, data);
+        };
+        (Some(Box::new(Self(values))), data)
+    }
+}
+
+impl Operation for ExtendUnsortedOp {
+    fn operate(&self, input: Input, _: Input) {
+        input.int_set.extend_unsorted(self.0.iter().copied());
+        input.btree_set.extend(self.0.iter().copied());
+    }
+
+    fn size(&self, length: usize) -> usize {
+        (length.ilog2() as usize) * self.0.len()
+    }
+}
+
+/* ### Union ### */
+
+struct UnionOp();
+
+impl UnionOp {
+    fn parse_args(data: &[u8]) -> (Option<Box<dyn Operation>>, &[u8]) {
+        (Some(Box::new(Self())), data)
+    }
+}
+
+impl Operation for UnionOp {
+    fn operate(&self, a: Input, b: Input) {
+        a.int_set.union(b.int_set);
+        for v in b.btree_set.iter() {
+            a.btree_set.insert(*v);
+        }
+    }
+
+    fn size(&self, length: usize) -> usize {
+        // TODO(garretrieger): should be length a + length b
+        length
+    }
+}
+
+/* ### Intersect ### */
+
+struct IntersectOp();
+
+impl IntersectOp {
+    fn parse_args(data: &[u8]) -> (Option<Box<dyn Operation>>, &[u8]) {
+        (Some(Box::new(Self())), data)
+    }
+}
+
+impl Operation for IntersectOp {
+    fn operate(&self, a: Input, b: Input) {
+        a.int_set.intersect(b.int_set);
+        let mut intersected: BTreeSet<u32> =
+            a.btree_set.intersection(b.btree_set).copied().collect();
+        std::mem::swap(a.btree_set, &mut intersected);
+    }
+
+    fn size(&self, length: usize) -> usize {
+        // TODO(garretrieger): should be length a + length b
+        length
+    }
+}
+
+/* ### End of Ops ### */
+
+fn read_u8(data: &[u8]) -> (Option<u8>, &[u8]) {
+    if data.is_empty() {
+        return (None, data);
+    }
+    (Some(data[0]), &data[1..])
+}
+
+fn read_u32(data: &[u8]) -> (Option<u32>, &[u8]) {
+    if data.len() < 4 {
+        return (None, data);
+    }
+    (
+        Some(
+            ((data[0] as u32) << 24)
+                | ((data[1] as u32) << 16)
+                | ((data[2] as u32) << 8)
+                | (data[3] as u32),
+        ),
+        &data[4..],
+    )
+}
```

Comment:
I think you are re-inventing FontData + Cursor and should likely use them?

Reply:
using cursor sounds good to me.